atomic_reactor/utils/operator.py

```diff
@@ -734,6 +734,28 @@ def _replace_pullspecs_not_in_annotations(self, obj, k_or_i, replacements):
             # in replacement_pullspecs and no replacement will be done
             self._replace_unnamed_pullspec(obj, k_or_i, replacements)
 
+    def _mods_append(self, append_mods):
+        if not append_mods:
+            return
+        for mod in append_mods:
+            for key in append_mods[mod]:
+                item = append_mods[mod][key]
+                if isinstance(item, list):
+                    self.data[mod].setdefault(key, []).extend(item)
+                else:
+                    #  Raise something - we're expecting a list
+                    pass
+
+    def _mods_update(self, update_mods):
+        if not update_mods:
+            return
+        for mod in update_mods:
+            for key in update_mods[mod]:
+                if not self.data.get(mod):
```

Comment:
'spec' & 'metadata' are required in schema

Reply:
@rcerven; I don't understand your comment in this context.

If the existing CSV is incorrect, it should have already been caught elsewhere; and the modifications file is *not* a complete CSV... so...?

Reply:
you don't need if/else here, since 'mod' will be always in CSV

you can do just: 
self.data[mod][key] = update_mods[mod][key]

Reply:
@rcerven; So, you're saying that a modifications file will *always* have an 'update' section, which will *always* contain 'metadata' and 'spec' keys? The example provided in the spec doc is prefaced with the comment that the "...JSON file provided to OSBS via a URL might look something like this" - "**MIGHT**"

Reply:
you are already passing JUST update section operator_csv._mods_update(operator_csv_modifications.get('update'))

'mod' is now spec/metadata

Reply:
derp. *If* an 'update' section is provided, any mods will *have to* include the full path, starting with 'metadata'/'spec' etc. Not sure what I was thinking...